### ISSUE TITLE
Eliminar columna de geometría al cargar una capa

### ISSUE THREAD

**Issue by erikasv** [PEM-Humboldt/dev-scrum#148]
En el servidor de la herramienta, después de cargar la capa a PostGIS se debe eliminar la columna de geometria (`geom`) de la tabla que se crea al cargarla. Esto porque Jorge nos manifestó que la existencia de esa columna hace más lenta la carga de los datos en el modelo (además es una columna que no se usa).

### Consideraciones
Se recomienda buscar una forma de eliminar esta columna que no implique agregar un orm o query builder a las dependencias del repositorio.

**Comment by erikasv** [comment]
@cazapatamar @Trjegul84 @danflop @alegnaaived Por favor volver a votar, esta tarea cambió del todo debido a que la versión anterior se hizo obsoleta.

Por favor si no tienen idea no votes y eliminense de la lista

**Comment by erikasv** [comment]
@danflop @alegnaaived Hice la prueba de publicar una capa desde postgis en el geoserver y me topé con un inconveniente: el sistema de coordenadas que se le estaba asignando a la capa cuando se carga a la base de datos es incorrecto, esto hace que la capa no se pueda visualizar bien (se le estaba pasando el sistema de la primera capa que nos dieron, pero el de la última era diferente).

Cuando la capa se lee desde un archivo, el geoserver verifica este sistema de coordenadas de otra forma, nosotros no tenemos que pasárselo; y el modelo sólo utiliza los datos alfanuméricos, así que le da igual si el sistema de coordenadas está bien o no.

Otra ventaja que le veo al manejo en archivos es que los usuarios finales pueden tener en el geoserver varias capas para consultarlas, mientras que si se lee desde la base de datos sólo podrían consultar la última que cargaron.

Creo que podemos dejar el manejo de las capas como está ahora y dedicar el tiempo restante a agregar el parametro al que corresponde la capa (`coberturas`) de acuerdo a lo último que hablamos con Jorge. ¿Qué opinan?

**Comment by alegnaaived** [comment]
Estoy de acuerdo en detener la tarea hasta aquí y en agregar el parámetro de la capa.

Pienso que podemos revisar luego si se puede configurar un sistema de coordenadas especial al arrancar el Geoserver o si se puede cambiar el sistema de coordenadas por defecto para que coincida con el de la capa en PostGIS.

**Comment by erikasv** [comment]
> Pienso que podemos revisar luego si se puede configurar un sistema de coordenadas especial al arrancar el Geoserver o si se puede cambiar el sistema de coordenadas por defecto para que coincida con el de la capa en PostGIS.

No entiendo cómo esto solucionaría el problema, el de la base de datos es el que está mal, además el sistema de coordenadas es por capa, no para todo el GeoServer:
> ... el sistema de coordenadas que se le estaba asignando a la capa cuando se carga a la base de datos es incorrecto...

**Comment by alegnaaived** [comment]
Lo había entendido al revés. ¿Se podría cambiar entonces el sistema de coordenadas al cargar la capa? @erikasv hasta el momento encontré [esta explicación](https://postgis.net/workshops/postgis-intro/projection.html#transforming-data), que implicaría ejecutar un script al cargar una capa. ¿Lo ves viable?

**Comment by danflop** [comment]
Creo que lo mejor entonces es descartar la solución de tener la capa sólo en la BD y continuar manejando el archivo en Geoserver y los datos requeridos en la BD.

**Comment by erikasv** [comment]
@alegnaaived la solución que planteas requiere conocer el sistema de coordenadas correcto al que se desea transformar los datos, pero para eso simplemente los cargaríamos con ese sistema de coordenadas desde el principio. El problema es que nosotros no sabemos en qué sistema de coordenadas viene la capa, por eso el usuario la tiene que ingresar, la herramienta que usamos para cargarla a la base de datos lo exige también. En cuanto al GeoServer no sé como identifica ese sistema cuando se carga la capa desde el archivo, seguro está en alguno de los archivos del zip, pero la herramienta de la base de datos no lo consulta entonces. Por eso esta parte en el primer comentario:

> Cuando la capa se lee desde un archivo, el geoserver verifica este sistema de coordenadas de otra forma, nosotros no tenemos que pasárselo..

Así como yo haciendo pruebas ingresé mal el sistema de coordenadas, podría pasar que el usuario lo ingrese mal; esto no afectaría el modelo pero sí la visualización, por eso propuse que siguiéramos manejando los archivos con el GeoServer, además del otro beneficio que luego identifiqué y que puede llegar a ser de mayor usabilidad para los usuarios.

---

Creo que con el comentario de @danflop podemos cerrar esta tarea y voy a crear otra pequeña para lo que estaba pensada inicialmente, eliminar la columna de la geometría y agregar el parámetro para identificar el módulo del modelo al que apunta la capa que se va a cargar.
